src/Adapters/Secondary/MySqlAdapter/Repositories/ProductRepository.ts

```diff
@@ -0,0 +1,109 @@
+import { Repository } from 'typeorm';
+import Product from '@Entities/Product'
+import IProductRepository from '@Application/Ports/Secondary/IProductRepository'
+import { Either, Left, Right } from '@Shared/util/either'
+import { AppDataSource } from '../index';
+import { Product as model } from '../models/Product';
+import NotFoundException from '../../../../Application/domain/Exceptions/NotFoundException';
+
+export default class ProductRepository implements IProductRepository {
+    private repository: Repository<model>
+    
+    constructor(){
+      this.repository = AppDataSource.getRepository(model)
+    };
+
+    async create(product: Product): Promise<Either<Error, string>> {
+      try {
+        const productModel = new model();
+        productModel.name = product.getName();
+        productModel.category = product.getCategory();
+        productModel.price = product.getPrice();
+        productModel.description = product.getDescription();
+        
+        await this.repository.save(productModel);
+
+        return Right<string>('Product has been created')
+      } catch (error) {
+        return Left<Error>(error as Error);
+      }
+    }
+
+    async update(product: Product): Promise<Either<Error, string>> {
+      try {
+        const productToUpdate = await this.repository.findOneBy({
+          id: product.getId(),
+        })
+
+        if (!productToUpdate) {
+          return Left<Error>(new NotFoundException('Product: ' + product.getId()))
+        }
+
+        productToUpdate.name = product.getName();
+        productToUpdate.category = product.getCategory();
+        productToUpdate.price = product.getPrice();
+        productToUpdate.description = product.getDescription();
+        
+        await this.repository.save(productToUpdate);
+
+        return Right<string>('Product has been updated')
+      } catch (error) {
+        return Left<Error>(error as Error);
+      }
+    }
+
+    async delete(id: string): Promise<Either<Error, string>> {
+      const productToRemove = await this.repository.findOneBy({
+        id,
+      })
+
+      if (!productToRemove) {
+        return Left<Error>(new NotFoundException('Product: ' + id))
+      }
+
+      await this.repository.remove(productToRemove);
+
+      return Right<string>('Product has been removed')
```

Comment:
Question (non-blocking): Try catch use

Why your not using trycatch in this block? is there some specific reason or it's just useless?

Reply:
Porque está apenas chamando o método do repository que está inteiramente coberto pelo try catch, então o erro é só repassado pelo either conforme o contrato